# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
# Commit: eb376fd954e1
# Full Hash: eb376fd954e1287b8e4f504f743b7159f1b514d0
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Change MediaEngineSource::SetTrack to take a MediaTrack base class. r=padenot
#   
#   This allows us to pass in different sub classes, which the next patch will do.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95933
# ==============================================================================

diff -r 2799805eef3b -r eb376fd954e1 dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
--- a/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp	Thu Nov 05 16:41:45 2020 +0000
+++ b/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp	Thu Nov 05 16:42:07 2020 +0000
@@ -263,14 +263,15 @@
   return NS_OK;
 }
 
-void MediaEngineRemoteVideoSource::SetTrack(
-    const RefPtr<SourceMediaTrack>& aTrack, const PrincipalHandle& aPrincipal) {
+void MediaEngineRemoteVideoSource::SetTrack(const RefPtr<MediaTrack>& aTrack,
+                                            const PrincipalHandle& aPrincipal) {
   LOG("%s", __PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
   MOZ_ASSERT(mState == kAllocated);
   MOZ_ASSERT(!mTrack);
   MOZ_ASSERT(aTrack);
+  MOZ_ASSERT(aTrack->AsSourceTrack());
 
   if (!mImageContainer) {
     mImageContainer = layers::LayerManager::CreateImageContainer(
@@ -279,7 +280,7 @@
 
   {
     MutexAutoLock lock(mMutex);
-    mTrack = aTrack;
+    mTrack = aTrack->AsSourceTrack();
     mPrincipal = aPrincipal;
   }
 }